pulp_container/app/tasks/sync_stages.py

```diff
@@ -287,9 +288,21 @@ async def resolve_flush(self):
         self.manifest_dcs.clear()
 
         for manifest_list_dc in self.manifest_list_dcs:
+            manifest_list_type = MANIFEST_TYPE.UNKNOWN
             for listed_manifest in manifest_list_dc.extra_data["listed_manifests"]:
                 # Just await here. They will be associated in the post_save hook.
                 await listed_manifest["manifest_dc"].resolution()
+
+                # if manifestlist type is not defined as "unknown" we already checked its nature,
+                # in this case, there is no need to verify the other manifests from list
+                if manifest_list_type != MANIFEST_TYPE.UNKNOWN and listed_manifest[
```

Comment:
This `if` condition is always evaluated to `False`, right? What is the rationale behind this procedure?

Reply:
> This if condition is always evaluated to False, right?

Right, but for the first iteration execution only. For the following iterations, the `manifest_list_type` may have been changed to `flatpak` or `bootable`.


> What is the rationale behind this procedure?

This is to avoid having to define the `manifest_list_type` in every iteration of the loop. If we already identified it as `flatpak` or `bootable`, we can skip checking the `listed_manifest["manifest_dc"].content.type`.

Reply:
Actually, you are right! My bad! :man_facepalming: 
The condition should be `if manifest_list_type == MANIFEST_TYPE.UNKNOWN and ...`